ArraysAndStrings/rotateMatrix/solution1b.js

```diff
@@ -0,0 +1,91 @@
+// rotate NxN matrix 90 degrees in place
+// O(N**2) time, O(1) space
+
+function rotateMatrix(matrix) {
+  const n = matrix.length
+  let depth = n / 2,
+    row = 0,
+    start = 0,
+    end = n - 1
+
+  while (row < depth) {
+    for (let col = start; col < end; col++) {
+      matrix = rotateFour(matrix, n, row, col)
+    }
+    row++
+    start++
+    end--
+  }
+
+  return matrix
+}
+
+function rotateFour(matrix, n, row, col) {
+  let tmp = matrix[row][col]
+
+  // (option A) rotate clockwise
+  for (let i = 0; i < 3; i++) {
+    const prevRow = n - 1 - col
+    const prevCol = row
+    matrix[row][col] = matrix[prevRow][prevCol]
+    row = prevRow
+    col = prevCol
+  }
+
+  // (option B) rotate counter-clockwise
```

Comment:
@jiannapark 

I like the idea of this second option, but wondering if you either want to:

a) actually implement that option in your function somehow or 
b) get rid of the comments
c) leave it as it is for a reason that you know better than me

Reply:
I thought it was cool to compare the patterns between option A and B, and depending on individual preference you can comment option A instead of B! Do you think it'd be neater to create a separate (solution1c.js) file for the second option?

Reply:
It is cool. 

Personally, I would just include it as an option in using the function. Right now rotate matrix takes a matrix. But since you have left and right options, why not put a parameter that you can pass that sets the direction and just let the user choose which to do?

Moreover, the option to set the mutliple of 90 degrees (so user could pass in 90, 180, 270) and get the result from either left or right.

So something like rotateMatrix(matrix, direction, degrees). 

But that's more work, so simply (matrix, direction) seems reasonable to me. I think the function can be neatly handled to provide that in one function rather than two solutions?

As I said, I also am happy to just merge this, because I don't know the rules we're really playing by here, just trying to say something thoughtful. :)